builtin/aws/ecs/platform.go

```diff
@@ -1057,6 +1148,71 @@ func (p *Platform) resourceAlbListenerCreate(
 	return nil
 }
 
+func (p *Platform) resourceAlbRedirectListenerDestroy(
+	ctx context.Context,
+	sg terminal.StepGroup,
+	sess *session.Session,
+	log hclog.Logger,
+	state *Resource_Alb_RedirectListener,
+) error {
+	if state.Arn == "" {
+		log.Debug("Missing alb redirect listener ARN - it must not have been created successfully. Skipping delete.")
+		return nil
+	}
+
+	s := sg.Add("Initiating deletion of ALB Redirect Listener (ARN: %q)", state.Arn)
+	defer s.Abort()
+
+	elbsrv := elbv2.New(sess)
+	s.Update("Describing redirect ALB listener (ARN: %q)", state.Arn)
+
+	listeners, err := elbsrv.DescribeListenersWithContext(ctx, &elbv2.DescribeListenersInput{
+		ListenerArns: []*string{&state.Arn},
+	})
+	if err != nil {
+		// There doesn't seem to be an aws error to cast to for this code.
+		if strings.Contains(err.Error(), "ListenerNotFound") {
+			s.Update("Redirect listener does not exist and must have been destroyed (ARN %q).", state.Arn)
```

Comment:
Probably will be better to re-phrase to:
`Redirect listener does not exist and it should have been destroyed`